### PR TITLE
Complete transfers with PSBT construction and tapret commitments

### PR DESCRIPTION
**Pull request by dr-orlovsky**

### CODE REVIEW

**Comment by dr-orlovsky** [comment]
Ok, I have finished implementing all functionality required to do a full transfer, from coin selection (including both co-operation bitcoin and RGB coin selection), PSBT creation, DBC commitments (with possible combination of oprets/taprets, which are automatically deduced from the inputs) to the consignment creation - all in just a single method call (and in single cli command `transfer`):

```rust
    let (psbt, meta, transfer) = rgb.pay(invoice, method, params)
```

I have no idea whether it works (pretty sure there are some bugs) - I will be debugging during the next few days. But at least, with the code written and compiling now I am sure we have everything we need in all our libraries (there are around 20 of them involved here) to have **real RGB taproot wallets** - a journey which took 3 years of my time and required implementing even taproot itself, twice (once in `rust-bitcoin`, second time in `bp-std`)...

File: cli/src/command.rs [hunk 1]
```diff
@@ -63,138 +65,152 @@ pub enum Command {
...
         value: u64,
     },
 
-    /// Create new transfer.
+    /// Transfer RGB assets
     #[display("transfer")]
     Transfer {
+        /// Encode PSBT as V2
+        #[clap(short = '2')]
+        v2: bool,
```
Comment (cryptoquick) [inline cli/src/command.rs:145]: Should there be a default value for this?

**Review by cryptoquick — COMMENTED** [review]
Very cool... Just one comment

File: cli/src/command.rs (hunk 1 above)
Comment (dr-orlovsky) [reply cli/src/command.rs:145]: it is a flag, thus "default value" is `false` assigned when the flag is absent

**Review by cryptoquick — APPROVED** [review]
ACK

**Pull request merged**